repose-aggregator/components/filters/openstack-identity-v3/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala

```diff
@@ -134,36 +134,38 @@ class OpenStackIdentityV3Handler(identityConfig: OpenstackIdentityV3Config, iden
         // Set the appropriate headers
         requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_TOKEN_EXPIRES, token.get.expires_at)
         requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_AUTHORIZATION.toString, OpenStackIdentityV3Headers.X_AUTH_PROXY) // TODO: Add the project ID if verified
-        token.get.user.name.map(requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_USER_NAME.toString, _))
-        token.get.roles.map { roles =>
+        token.get.user.name.foreach(requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_USER_NAME.toString, _))
+        token.get.roles.foreach { roles =>
           requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_ROLES, roles.map(_.name) mkString ",")
         }
-        token.get.user.id.map { id =>
+        token.get.user.id.foreach { id =>
           requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_USER_ID.toString, id)
           requestHeaderManager.appendHeader(PowerApiHeader.USER.toString, id, 1.0)
         }
-        token.get.user.rax_default_region.map {
-          requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_DEFAULT_REGION.toString, _)
+        token.get.user.rax_default_region.foreach { defaultRegion =>
+          requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_DEFAULT_REGION.toString, defaultRegion)
         }
-        if (identityConfig.isSendAllProjectIds) {
-          writeProjectHeader(token.flatMap(_.project.flatMap(_.id)).orNull, token.flatMap(_.roles).orNull, writeAll = true, filterDirector)
-        } else {
-          projectIdUriRegex match {
-            case Some(regex) => writeProjectHeader(extractProjectIdFromUri(projectIdUriRegex.get, request.getRequestURI).orNull, token.flatMap(_.roles).get, writeAll = false, filterDirector)
-            case None if token.flatMap(_.project).flatMap(_.id).isDefined =>
-              token flatMap (_.project) map { project =>
-                project.id.map(requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_PROJECT_ID.toString, _))
-                project.name.map(requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_PROJECT_NAME.toString, _))
-              }
-            case None => None
+        token.get.project.foreach { project =>
+          project.name.foreach { projectName =>
+            requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_PROJECT_NAME.toString, projectName)
           }
         }
-        token.get.rax_impersonator.map { impersonator =>
-          impersonator.id.map(requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_IMPERSONATOR_ID.toString, _))
-          impersonator.name.map(requestHeaderManager.putHeader(OpenStackIdentityV3Headers.X_IMPERSONATOR_NAME.toString, _))
+        projectIdUriRegex match {
+          case Some(regex) =>
+            writeProjectHeader(token.flatMap(_.project.flatMap(_.id)), token.flatMap(_.roles).getOrElse(List[Role]()),
```

Comment:
It was kind of like that before... though I did have to add another method argument.

I can pull the flapMap calls out and stick them into vals for readability. Same with the extractProjectIdFromUri(...) call. Would that be enough?

Part of the problem is that we pass config values to make testing easier, but there really isn't a requirement that they be passed. The config is a field, after all.